### PR TITLE
Prevent double-decode of url param. See #9 and #10

### PR DESCRIPTION
**Pull request by xxx**
Do not decode url param a second time before converting. Causes normalization to be more painful than it needs to be, and the url is already decoded at this point.

Normalization is necessary because cURL does not do it itself.

### CODE REVIEW

**Comment by xxx** [comment]
See also #9 and #10

**Comment by sdepold** [comment]
what a pity that u put everything into once pull request. will merge it now

File: lib/request-handler.js [hunk 1]
```diff
@@ -138,7 +138,7 @@ RequestHandler.prototype._afterResize = function(err, path, res, callback) {
     res.contentType(path.replace(/\?(.*)/, ""))
     res.sendfile(path, { maxAge: oneYear }, function() {
       self.logger.log('File sent')
-      self.deleteTempfile(path)
+//      self.deleteTempfile(path)
```
Comment (sdepold) [inline lib/request-handler.js:141]: what's up with this line?

**Comment by sdepold** [comment]
thanks a lot :) nice work

File: lib/request-handler.js (hunk 1 above)
Comment (xxx) [inline lib/request-handler.js:141]: I changed it so the converted file is cached and served if a request comes in to transform the file the same way, to the same size.

Comment (sdepold) [inline lib/request-handler.js:141]: right. now i see :) didn't notice that you pass the actual url to that method!

Comment (sdepold) [inline lib/request-handler.js:141]: fyi: we just had a major fuckup as the hdds of our ec2 instances were out of space. I will add an options, which enables one to keep the downloaded files. also i will add an option for a limit of max used tmp space.

Comment (xxx) [inline lib/request-handler.js:141]: ack, sorry to hear that. i threw a terabyte at ours so disk space isn't an issue, but the load we're seeing is causing a lot of nodejs restarts

Comment (sdepold) [inline lib/request-handler.js:141]: what do u mean by "causing a lot of nodejs restarts" ? we have an haproxy in front of our instances and are queueing requests if too many are approaching.

Comment (xxx) [inline lib/request-handler.js:141]: It's due to memory usage in the node process. We're also using haproxy for request queueing

**Pull request merged**